quinn/src/connection.rs

```diff
@@ -59,8 +60,14 @@ impl Connecting {
             runtime.clone(),
         );
 
+        let driver = ConnectionDriver(conn.clone());
         runtime.spawn(Box::pin(
-            ConnectionDriver(conn.clone()).instrument(Span::current()),
+            async {
+                if let Err(e) = driver.await {
+                    tracing::error!("I/O error: {e}");
+                }
+            }
+            .instrument(Span::current()),
```

Comment:
It feels like this commit has a lot going on, and could be split into smaller parts?

Reply:
Split into three parts:
- Adding an error path to `ConnectionDriver`
- Moving transmits into `ConnectionDriver`
- Removing `quinn::EndpointEvent`

Is that clearer? Neither of the newly split out commits are justified in isolation, but they do help isolate the actual behavior change.